src/gis/components/Map.js

```diff
@@ -95,146 +105,171 @@ const LeafletSearch = props => {
-    () => ({
-      dragend: () => {
-        const marker = markerRef.current;
-        if (marker != null) {
-          setPinLocation(marker.getLatLng());
-        }
+    geojsonToLayer(geojson, featureGroup, {
+      pointToLayer: (feature, latlng) => {
+        const marker = L.marker(latlng, { draggable: true });
+        marker.on('dragend', event => {
+          console.log('dragend', { event });
+          setNewGeoJson(layerToGeoJSON(featureGroup.getLayers()));
+        });
+        return marker;
       },
-    }),
-    []
-  );
+    });
+  }, [featureGroup, geojson, onGeoJsonChange]);
 
   useEffect(() => {
-    if (pinLocation && boundingBox) {
-      onPinLocationChange({
-        pinLocation,
-        boundingBox,
-      });
+    if (!newGeoJson) {
+      return;
     }
-  }, [boundingBox, pinLocation, onPinLocationChange]);
+    console.log('update geojson');
+
+    onGeoJsonChange(newGeoJson);
+  }, [newGeoJson, onGeoJsonChange]);
+
+  return null;
+};
+
+const Location = props => {
+  const { enableSearch, enableDraw } = props;
 
-  useEffect(() => {
-    const getZoomData = () => {
-      const southWest = map.getBounds().getSouthWest();
-      const northEast = map.getBounds().getNorthEast();
-      const bbox = [southWest.lng, southWest.lat, northEast.lng, northEast.lat];
-      if (bbox) {
-        setBoundingBox(bbox);
-      }
-    };
-    getZoomData();
-    map.on('zoomend', getZoomData);
-  }, [map]);
   return (
     <>
-      {enableSearch && (
-        <LeafletSearch
-          setPinLocation={setPinLocation}
-          setBoundingBox={setBoundingBox}
-        />
-      )}
-
-      {enableDraw && (
-        <LeafletDraw
-          setPinLocation={setPinLocation}
-          setBoundingBox={setBoundingBox}
-        />
-      )}
-
-      {pinLocation && (
-        <Marker
-          draggable
-          ref={markerRef}
-          position={pinLocation}
-          eventHandlers={markerEventHandlers}
-        />
-      )}
+      {enableSearch && <LeafletSearch />}
+      {enableDraw && <LeafletDraw />}
     </>
   );
 };
 
 const Map = props => {
   const [map, setMap] = useState();
+  const [featureGroup, setFeatureGroup] = useState();
   const isSmall = useMediaQuery(theme.breakpoints.down('sm'));
+  const { bounds, geojson, onGeoJsonChange } = props;
 
   useEffect(() => {
     if (map?.target && props.center) {
       map.target.flyTo(props.center, 3);
     }
   }, [props.center, map]);
 
+  useEffect(() => {
+    if (bounds) {
+      map?.target.fitBounds(bounds);
+    }
+  }, [map?.target, bounds]);
+
+  useEffect(() => {
+    if (map?.target) {
+      const featureGroup = L.featureGroup().addTo(map?.target);
+      setFeatureGroup(featureGroup);
+    }
+  }, [map?.target]);
+
+  const updateBounds = useCallback(
+    bbox => {
+      if (!onGeoJsonChange) {
+        return;
+      }
+      onGeoJsonChange(current => {
+        if (!current) {
+          return current;
+        }
```

Comment:
suggestion: This seems awkward. Can you return false or null (or just `return`) instead?